bson/encoder_example_test.go

```diff
@@ -121,3 +122,126 @@ func ExampleEncoder_UseJSONStructTags() {
 	fmt.Println(bson.Raw(buf.Bytes()).String())
 	// Output: {"name": "Cereal Rounds","sku": "AB12345","price_cents": {"$numberLong":"399"}}
 }
+
+func ExampleEncoder_multipleBSONDocuments() {
+	// Create an Encoder that writes BSON values to a bytes.Buffer.
+	buf := new(bytes.Buffer)
+	vw, err := bsonrw.NewBSONValueWriter(buf)
+	if err != nil {
+		panic(err)
+	}
+	encoder, err := bson.NewEncoder(vw)
+	if err != nil {
+		panic(err)
+	}
+
+	type Coordinate struct {
+		X int
+		Y int
+	}
+
+	// Use the encoder to marshal 5 Coordinate values as a sequence of BSON
+	// documents.
+	for i := 0; i < 5; i++ {
+		err := encoder.Encode(Coordinate{
+			X: i,
+			Y: i,
+		})
+		if err != nil {
+			panic(err)
+		}
+	}
+
+	// Read each marshaled BSON document from the buffer and print them as
+	// Extended JSON by converting them to bson.Raw.
+	for {
+		doc, err := bson.ReadDocument(buf)
+		if err == io.EOF {
+			return
+		}
+		if err != nil {
+			panic(err)
+		}
+		fmt.Println(bson.Raw(doc).String())
+	}
+	// Output:
+	// {"x": {"$numberInt":"0"},"y": {"$numberInt":"0"}}
+	// {"x": {"$numberInt":"1"},"y": {"$numberInt":"1"}}
+	// {"x": {"$numberInt":"2"},"y": {"$numberInt":"2"}}
+	// {"x": {"$numberInt":"3"},"y": {"$numberInt":"3"}}
+	// {"x": {"$numberInt":"4"},"y": {"$numberInt":"4"}}
+}
+
+func ExampleEncoder_extendedJSON() {
+	// Create an Encoder that writes canonical Extended JSON values to a
+	// bytes.Buffer.
+	buf := new(bytes.Buffer)
+	vw, err := bsonrw.NewExtJSONValueWriter(buf, true, false)
+	if err != nil {
+		panic(err)
+	}
+	encoder, err := bson.NewEncoder(vw)
+	if err != nil {
+		panic(err)
+	}
+
+	type Product struct {
+		Name  string `bson:"name"`
+		SKU   string `bson:"sku"`
+		Price int64  `bson:"price_cents"`
+	}
+
+	// Use the Encoder to marshal a BSON document that contains the name, SKU,
+	// and price (in cents) of a product.
+	product := Product{
+		Name:  "Cereal Rounds",
+		SKU:   "AB12345",
+		Price: 399,
+	}
+	err = encoder.Encode(product)
+	if err != nil {
+		panic(err)
+	}
+
+	fmt.Println(buf.String())
+	// Output: {"name":"Cereal Rounds","sku":"AB12345","price_cents":{"$numberLong":"399"}}
+}
+
+func ExampleEncoder_multipleExtendedJSONDocuments() {
+	// Create an Encoder that writes canonical Extended JSON values to a
+	// bytes.Buffer.
+	buf := new(bytes.Buffer)
+	vw, err := bsonrw.NewExtJSONValueWriter(buf, true, false)
+	if err != nil {
+		panic(err)
+	}
+	encoder, err := bson.NewEncoder(vw)
+	if err != nil {
+		panic(err)
+	}
+
+	type Coordinate struct {
+		X int
+		Y int
+	}
+
+	// Use the encoder to marshal 5 Coordinate values as a sequence of Extended
+	// JSON documents.
+	for i := 0; i < 5; i++ {
+		err := encoder.Encode(Coordinate{
+			X: i,
+			Y: i,
```

Comment:
Could we give these fields different values, imposing a stricter test example?

```suggestion
			Y: i+1,
```